core/backend/src/IModelDb.ts

```diff
@@ -1997,13 +2024,30 @@ export namespace IModelDb {
      * @note Useful for cases when an element may or may not exist and throwing an `Error` would be overkill.
      * @see getElementProps
      */
-    public tryGetElementProps<T extends ElementProps>(elementId: Id64String | GuidString | Code | ElementLoadProps): T | undefined {
-      if (typeof elementId === "string") {
-        elementId = Id64.isId64(elementId) ? { id: elementId } : { federationGuid: elementId };
-      } else if (elementId instanceof Code) {
-        elementId = { code: elementId };
+    public tryGetElementProps<T extends ElementProps>(props: Id64String | GuidString | Code | ElementLoadProps): T | undefined {
+      if (typeof props === "string") {
+        props = Id64.isId64(props) ? { id: props } : { federationGuid: props };
+      } else if (props instanceof Code) {
+        props = { code: props };
+      }
+      try {
+        if (IModelHost.configuration?.enableThinnedNativeInstanceWorkflow) {
+          const cachedElm = this.cache.get(props);
+          if (cachedElm) {
+            return cachedElm.elProps as T;
+          }
+          const options = { ...props, useJsNames: true };
+          const instanceKey = this.resolveElementKey(props);
+          const rawInstance = this._iModel[_nativeDb].readInstance(instanceKey, options) as ECSqlRow;
+          const classDef = this._iModel.getJsClass<typeof Element>(rawInstance.classFullName);
+          const elementProps = classDef.deserialize({ row: rawInstance, iModel: this._iModel, options: { element: props } }) as T;
+          this.cache.set({ elProps: elementProps, loadOptions: props });
+          return elementProps;
+        }
+        return this._iModel[_nativeDb].getElement(props) as T;
```

Comment:
```suggestion
        if (!IModelHost.configuration?.enableThinnedNativeInstanceWorkflow) {
          return this._iModel[_nativeDb].getElement(props) as T;
        }
        
        const cachedElm = this.cache.get(props);
        if (cachedElm) {
          return cachedElm.elProps as T;
        }

        const options = { ...props, useJsNames: true };
        const instanceKey = this.resolveElementKey(props);
        const rawInstance = this._iModel[_nativeDb].readInstance(instanceKey, options) as ECSqlRow;
        const classDef = this._iModel.getJsClass<typeof Element>(rawInstance.classFullName);
        const elementProps = classDef.deserialize({ row: rawInstance, iModel: this._iModel, options: { element: props } }) as T;
        this.cache.set({ elProps: elementProps, loadOptions: props });
        return elementProps;
```

Nit but this is way easier to understand IMO.

Reply:
Committing, changing getModel() to reflect this as well

Reply:
applied